docs/examples/gen_velocities.rst

```diff
@@ -0,0 +1,432 @@
+      -
+
+    * - **Further parameters**
+      -  
+      -
+
+    * - distance to the pulsar system
+      - :math:`d_\mathrm{p}`
+      -
+
+    * - distance to the screen
+      - :math:`d_\mathrm{s}`
+      - from Earth
+
+    * - position angle of the lens
+      - :math:`\mathit{\Omega}_\mathrm{s}`
+      - measured from north through east to the direction of motion on the sky
+
+    * - velocity of the interstellar medium
+      - :math:`v_\mathrm{ISM}`
+      - projected onto the screen direction
+
+.. plot::
+    :include-source:
+    :context:
+
+    p_b = 5.7410459 * u.day
+    asini_p = 3.3667144 * const.c * u.s
+    i_p = 137.56 * u.deg
+    omega_p = 207. * u.deg
+    t0_p = Time(54501.4671, format='mjd')
+
+    d_p = 156.79 * u.pc
+    d_s = 90.6 * u.pc
+    omega_s = 134.6 * u.deg
+    v_ism = -31.9 * u.km / u.s
+
+The coordinates should be placed directly in a
+:py:class:`~astropy.coordinates.SkyCoord` object, that includes the pulsar
+system's position on the sky, its distance, and its proper motion.
+
+.. plot::
+    :include-source:
+    :context:
+
+    psr_coord = SkyCoord('04 37 15.99744 -47 15 09.7170',
+                         unit=(u.hourangle, u.deg),
+                         distance=d_p,
+                         pm_ra_cosdec=121.4385 * u.mas / u.yr,
+                         pm_dec=-71.4754 * u.mas / u.yr)
+
+Calculate some derived quantities:
+
+.. list-table::
+    :widths: 2 1
+    :header-rows: 1
+
+    * - Parameter
+      - Equation
+
+    * - semi-amplitude of the pulsar's projected velocity
+      - 
+        .. math::
+            
+            K_\mathrm{p} = \frac{ 2 \pi a_\mathrm{p} \sin( i_\mathrm{p} ) }
+                                { P_\mathrm{b} }
+
+    * - fractional distance to the screen (from Earth)
+      - 
+        .. math::
+            
+            s = 1 - \frac{ d_\mathrm{s} }{ d_\mathrm{p} }
+
+    * - effective distance
+      - 
+        .. math::
+        
+            d_\mathrm{eff} = \frac{ d_\mathrm{p} d_\mathrm{s} }
+                                  { d_\mathrm{p} - d_\mathrm{s} }
+
+    * - angle from the lens to the pulsar orbit's line of nodes
+      - 
+        .. math::
+        
+            \mathit{\Delta\Omega}_\mathrm{p}
+              = \mathit{\Omega}_\mathrm{s} - \mathit{\Omega}_\mathrm{p}
+
+.. plot::
+    :include-source:
+    :context:
+
+    k_p = 2.*np.pi * asini_p / p_b
+
+    s = 1 - d_s / d_p
+    d_eff = d_p * d_s / (d_p - d_s)
+
+    delta_omega_p = omega_s - omega_p
+
+Define a grid of times :math:`t` for which you want to calculate curvatures.
+
+.. plot::
+    :include-source:
+    :context:
+
+    t_mjd = np.arange(55000., 55700., 0.25)
+    t = Time(t_mjd, format='mjd')
+
+The lens frame
+==============
+
+Make a :py:class:`~astropy.coordinates.SkyOffsetFrame` centered on the pulsar
+system, rotated to the one-dimensional lens.
+
+.. plot::
+    :include-source:
+    :context:
+
+    lens_frame = SkyOffsetFrame(origin=psr_coord, rotation=omega_s)
```

Comment:
Not sure if "lens frame" is entirely correct, because `psr_coord` includes the distance to the pulsar system.

Reply:
True, though the offset frame doesn't care about that and the proper motion. let's leave it for now...